django/db/models/base.py

```diff
@@ -1119,16 +1139,32 @@ def _save_table(
                 for f in meta.local_concrete_fields
                 if not f.generated and (pk_set or f is not meta.auto_field)
             ]
-            returning_fields = meta.db_returning_fields
+            returning_fields = list(meta.db_returning_fields)
+            for field in fields:
+                value = (
+                    getattr(self, field.attname) if raw else field.pre_save(self, False)
+                )
+                if hasattr(value, "resolve_expression"):
+                    returning_fields.append(field)
+                elif field.db_returning:
+                    returning_fields.remove(field)
```

Comment:
If an explicit literal value is provided for a field there is no need to return it.

Reply:
Except that I've written some code like this for MongodB where the id is [`ObjectIdAutoField`](https://github.com/mongodb/django-mongodb-backend/blob/4411a08cd3cfb50ab3753924db5935e0031a7510/django_mongodb_backend/fields/auto.py): `Question.objects.create(id="000000000000000000000013")`. Previously, this would refresh `id` to the returned value, `ObjectId("000000000000000000000013")`. What do you think?

Reply:
I guess the same problem might happen with some form of UUID field that returns `uuid.UUID` instances I guess. Could you write a regression test for it and file an issue?